File: app/master-data-management/src/main/java/unipos/data/components/taxRates/TaxRateService.java
package unipos.data.components.taxRates;

import java.util.List;

/**
 * Created by dominik on 28.07.15.
 */
public interface TaxRateService {
    List<TaxRate> findAll();

    void deleteAllTaxRates();

    void deleteByGuid(String guid);

    void saveTaxRate(TaxRate taxRate);

    void deleteTaxRateByMongoId(String dbId);

    void updateTaxRate(TaxRate taxRate);
}

File: app/master-data-management/src/main/webapp/components/taxRate/taxRateService.js
//# sourceURL=taxRateService.js
/**
 * Created by dominik on 04.08.15.
 */

define(["angular"], function (angular) {
    return ["$resource", function ($resource) {

        var taxRateResource = $resource('/data/taxRates', {}, {
            deleteTaxRateByGuid: {
                method: 'DELETE',
                isArray:false,
                url: '/data/taxRates/guid'
            }
        });

        this.findAll = function () {
                var taxRates = taxRateResource.query();
            return taxRates;
        };

        this.save = function(taxRate) {
            return taxRateResource.save(taxRate);
        }

        this.checkIfResourceInstance = function(instance) {
            return (instance instanceof taxRateResource);
        };

        this.update = function (taxRate) {
            return taxRate.$save();
        };

        this.save = function(taxRate) {
            return taxRateResource.save(taxRate);
        };

        this.deleteByGuid = function(guid) {
            return taxRateResource.deleteTaxRateByGuid({guid: guid});
        };
    }];
});